relay-general/src/store/normalize/contexts.rs

```diff
@@ -199,11 +204,45 @@ fn normalize_response(response: &mut ResponseContext) {
     }
 }
 
+// Reads device specs (family, memory, cpu, etc) and sets the device class to high, medium, or low.
+fn normalize_device_context(device: &mut DeviceContext) {
+    if let Some(family) = device.family.value() {
+        if family == "iPhone" || family == "iOS" || family == "iOS-Device" {
+            if let Some(processor_frequency) = device.processor_frequency.value() {
+                if processor_frequency < &2000 {
+                    device.class = Annotated::new(DEVICE_CLASS_LOW);
+                } else if processor_frequency < &3000 {
+                    device.class = Annotated::new(DEVICE_CLASS_MEDIUM);
+                } else {
+                    device.class = Annotated::new(DEVICE_CLASS_HIGH);
+                }
+            }
+        } else if device.processor_frequency.value().is_some()
+            && device.processor_count.value().is_some()
+            && device.memory_size.value().is_some()
+        {
+            if device.processor_frequency.value().unwrap() < &2000
+                || device.memory_size.value().unwrap() < &(4 * GIB)
+                || device.processor_count.value().unwrap() < &8
+            {
+                device.class = Annotated::new(DEVICE_CLASS_LOW);
+            } else if device.processor_frequency.value().unwrap() < &2500
+                || device.memory_size.value().unwrap() < &(6 * GIB)
+            {
+                device.class = Annotated::new(DEVICE_CLASS_MEDIUM);
+            } else {
+                device.class = Annotated::new(DEVICE_CLASS_HIGH);
+            }
+        }
```

Comment:
You can avoid these unwraps with an `if let` statement similar to the one above:

```suggestion
        } else if let (Some(&freq), Some(&proc), Some(&mem)) = (
            device.processor_frequency.value(),
            device.processor_count.value(),
            device.memory_size.value(),
        ) {
            if freq < 2000 || mem < 4 * GIB || proc < 8 {
                device.class = Annotated::new(DEVICE_CLASS_LOW);
            } else if freq < 2500 || mem < 6 * GIB {
                device.class = Annotated::new(DEVICE_CLASS_MEDIUM);
            } else {
                device.class = Annotated::new(DEVICE_CLASS_HIGH);
            }
        }
```

Reply:
👍 Thanks Jan, updated to if let!